### PR TITLE
MGMT-12167: Add ZTP job to assisted-installer repo

### PR DESCRIPTION
**Pull request by adriengentil**
- Assisted: run ztp job when controller pod definition changes
- assisted-baremetal-operator-setup: allow redefining ci-index image

### CODE REVIEW

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml [hunk 1]
```diff
@@ -189,6 +197,17 @@ tests:
...
+- as: e2e-ai-operator-ztp
+  optional: true
+  steps:
+    cluster_profile: packet-assisted
+    dependencies:
+      ASSISTED_OPENSHIFT_INSTALL_RELEASE_IMAGE: release:latest
+      INDEX_IMAGE: assisted-service-operator-index
+      OPENSHIFT_INSTALL_RELEASE_IMAGE: release:latest
+    env:
+      INDEX_IMAGE_OVERRIDE: quay.io/edge-infrastructure/assisted-service-index:latest
```
Comment (adriengentil) [inline ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:209]: It maybe better to add it in the supplemental images, what do you think @osherdp

Comment (adriengentil) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:209]: that way, we just have to redefine `INDEX_IMAGE`

Comment (adriengentil) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:209]: and cleanup all references to `assisted-service-operator-index` as it is broken (promoted from Prow's `ci-index`)

Comment (osherdp) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:209]: yeah it should be better than we have right now

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-release-ocm-2.6.yaml [hunk 2]
```diff
@@ -152,6 +160,15 @@ tests:
         SERVICE_BASE_REF=release-ocm-2.6
         NUM_MASTERS=1
     workflow: assisted-baremetal
+- as: e2e-ai-operator-ztp
+  optional: true
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-release-ocm-2.6.yaml:164]: for 2.6 branch I think it should be a required job

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-release-ocm-2.6.yaml [hunk 3]
```diff
@@ -152,6 +160,15 @@ tests:
         SERVICE_BASE_REF=release-ocm-2.6
         NUM_MASTERS=1
     workflow: assisted-baremetal
+- as: e2e-ai-operator-ztp
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-release-ocm-2.6.yaml:163]: can you apply ``skip_if_only_changed/run_if_changed`` configurations?

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml [hunk 4]
```diff
@@ -11,6 +11,14 @@ base_images:
     name: ocm-2.7
     namespace: edge-infrastructure
     tag: assisted-service
+  assisted-service-index:
+    name: assisted-service-index
+    namespace: edge-infrastructure
+    tag: latest
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: for conformity, should we stick with the ``ocm-2.7`` name convention?

Comment (osherdp) [inline ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: another question about this index, maybe more relevant to the previous PR of supplemental images
don't we have a similar mechanism for updating this index?
https://github.com/openshift/release/blob/0896191a7c5ec03f0bbb9038bc1fe8ac14f887a2/ci-operator/step-registry/assisted/baremetal/operator/catalog-publish/assisted-baremetal-operator-catalog-publish-ref.yaml#L36-L43

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml (hunk 4 above)
Comment (adriengentil) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: I'm not sure to understand your question, what do you mean by "don't we have a similar mechanism for updating this index" ? The `assisted-baremetal-operator-catalog-publish` step publishes the image directly to quay.

Comment (adriengentil) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: I kept the naming from what we deploy to quay, at least it stays consistent with its usage in assisted-test-infra: https://github.com/openshift/assisted-test-infra/blob/master/Makefile#L45

But maybe we should change that later?

Comment (osherdp) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: sure

Comment (osherdp) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: So ``assisted-baremetal-operator-catalog-publish`` publishes index to ``quay.io/edge-infrastructure/assisted-service-index:latest``, and the supplemental-images mirroring mirrors it to ``registry.ci.openshift.org/edge-infrastructure/assisted-service-index:latest``
In addition, here we're promoting the index to namespace ``edge-infrastructure`` and name ``assisted-service-index:latest`` (at least from what I understand) and promotion is automatically available at ``registry.ci.openshift.org``. How does those two paths of updating ``registry.ci.openshift.org/edge-infrastructure/assisted-service-index:latest`` play along?

Comment (osherdp) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: It's a bit technical, so if I'm just missing some detail here let me know

Comment (adriengentil) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: the broken index we promote from `ci-index` is accessible from `registry.ci.openshift.org/edge-infrastructure/ocm-2.7:assisted-service-operator-index`: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml#L160.

Their names do not collide.

Comment (osherdp) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: missed that
thanks!

**Comment by osherdp** [comment]
So looks good to me
Too bad it doesn't want to rehearse disconnected...

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-release-ocm-2.6.yaml (hunk 2 above)
Comment (adriengentil) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-release-ocm-2.6.yaml]: done

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-release-ocm-2.6.yaml (hunk 3 above)
Comment (adriengentil) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-release-ocm-2.6.yaml:163]: done for 2.6.
It's optional on master, so we don't need it

File: ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml (hunk 4 above)
Comment (adriengentil) [reply ci-operator/config/openshift/assisted-installer/openshift-assisted-installer-master__edge.yaml:17]: no problem !

**Comment by adriengentil** [comment]
> So looks good to me Too bad it doesn't want to rehearse disconnected...

now it triggered ! :partying_face:

**Comment by adriengentil** [comment]
:crossed_fingers: 
/test pj-rehearse-max

**Comment by adriengentil** [comment]
`e2e-ai-operator-ztp-disconnected` passed :partying_face:

File: ci-operator/jobs/openshift/assisted-installer/openshift-assisted-installer-master-presubmits.yaml [hunk 5]
```diff
@@ -1,5 +1,82 @@
 presubmits:
   openshift/assisted-installer:
+  - agent: kubernetes
+    always_run: true
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-installer/openshift-assisted-installer-master-presubmits.yaml:4]: is it intended to have it always running?

**Review by osherdp — COMMENTED** [review]
just one comment

**Review by osherdp — APPROVED** [review]

File: ci-operator/jobs/openshift/assisted-installer/openshift-assisted-installer-master-presubmits.yaml (hunk 5 above)
Comment (osherdp) [reply ci-operator/jobs/openshift/assisted-installer/openshift-assisted-installer-master-presubmits.yaml]: yes good idea :)

**Pull request merged**